react/features/stream-effects/rnnoise/RnnoiseProcessor.js

```diff
@@ -175,23 +133,33 @@ export default class RnnoiseProcessor {
      * Calculate the Voice Activity Detection for a raw Float32 PCM sample Array.
      * The size of the array must be of exactly 480 samples, this constraint comes from the rnnoise library.
      *
-     * @param {Float32Array} pcmFrame - Array containing 32 bit PCM samples.
+     * @param {Float32Array} pcmFrame - Array containing 32 bit PCM samples. Parameter is also used as output
+     * when {@code shouldDenoise} is true.
+     * @param {boolean} shouldDenoise - Should the denoised frame be returned in pcmFrame.
      * @returns {Float} Contains VAD score in the interval 0 - 1 i.e. 0.90 .
      */
-    calculateAudioFrameVAD(pcmFrame: Float32Array) {
-        if (this._destroyed) {
-            throw new Error('RnnoiseProcessor instance is destroyed, please create another one!');
+    calculateAudioFrameVAD(pcmFrame: Float32Array, shouldDenoise: Boolean = false) {
```

Comment:
Can we detach the functionality here? I find it very confusing that an API described as "Calculate the Voice Activity Detection for a raw Float32 PCM sample Array." also denoises. the audio.

Reply:
I've renamed the function to something a bit more suggestive